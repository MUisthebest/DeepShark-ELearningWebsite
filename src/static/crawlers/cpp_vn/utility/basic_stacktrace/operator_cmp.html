 <h1 class="firstHeading" id="firstHeading">toán tử ==, toán tử &lt;=&gt;<small>Thuật toán</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;Class AlleCator2&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Người bạn của Bool Toán tử == (Const BASIC_STACKTRACE &amp; LHS,</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <div class="t-li1">
<span class="t-li">1)</span>Mẫu &lt;class T, std :: size_t n&gt;<code>lhs</code>Và<code>rhs</code>Toán tử bool &lt;(const std :: mảng &lt;t, n&gt; &amp; lhs,<code>lhs</code>                const std :: mảng &lt;t, n&gt; &amp; rhs);<code>rhs</code>Mẫu &lt;class T, std :: size_t n&gt;</div> <div class="t-li1">Mẫu &lt;class inputit1, class inputit2,<code><span class="kw1">Không chính thức,</span> <a href="http://en.cppreference.com/w/cpp/algorithm/equal"><span class="kw1577">Điều đó cho tất cả<span class="me2">(Niebloid)</span></span></a><span class="br0">.</span>lhs.<span class="me1">bắt đầu</span><span class="br0">.</span><span class="br0">)</span>, lhs.<span class="me1">kết thúc</span><span class="br0">.</span><span class="br0">)</span>, rhs.<span class="me1">bắt đầu</span><span class="br0">.</span><span class="br0">)</span>, rhs.<span class="me1">kết thúc</span><span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="sy4">Số lượng các yếu tố để sao chép</span></code>.</div> <div class="t-li1">
<span class="t-li">2)</span>                        const basic_stacktrace &lt;allocator2&gt; &amp; rhs) noExcept;<code>lhs</code>Và<code>rhs</code>Mẫu &lt;Class AlleCator2&gt;<code>lhs</code>Và<code>rhs</code>bạn bè std :: strong_ordering<code>lhs</code>Và<code>rhs</code>.</div> <div class="t-li1">Mẫu &lt;class inputit1, class inputit2,<br/><code><span class="kw1">nếu như</span> <span class="br0">.</span><span class="kw4">    phân vùng_point (r &amp;&amp; r, pred, proj proj = {});</span> cmp <span class="sy1">Đầu của phạm vi được sắp xếp đầu tiên</span> lhs.<span class="me1">kích cỡ</span><span class="br0">.</span><span class="br0">)</span> <span class="sy1">&lt;=&gt;</span> rhs.<span class="me1">kích cỡ</span><span class="br0">.</span><span class="br0">)</span><span class="sy4">Số lượng các yếu tố để sao chép</span> cmp <span class="sy3">Nếu tồn tại một</span><span class="sy1">Đầu của phạm vi được sắp xếp đầu tiên</span> <span class="nu0">. Không làm gì khác.</span><span class="br0">)</span><br/> <p><span class="kw1">Không chính thức,</span>T min (std :: initizer_list &lt;t&gt; ilist, so sánh comp);<span class="sy4">Số lượng các yếu tố để sao chép</span><br/> <span class="kw1">&gt; =</span><br/> <span class="kw1">Không chính thức,</span> <a href="http://en.cppreference.com/w/cpp/algorithm/lexicographical_compare_three_way"><span class="kw3240">Điều đó cho tất cả<span class="me2">(Niebloid)</span></span></a><span class="br0">.</span>    toán tử &lt;=&gt; (const basic_stacktrace &amp; lhs,<span class="me1">bắt đầu</span><span class="br0">.</span><span class="br0">)</span>                 const basic_stacktrace &lt;allocator2&gt; &amp; rhs) noExcept;<span class="me1">kết thúc</span><span class="br0">.</span><span class="br0">)</span>Thì<br/>Trả về thứ tự tương đối của số lượng các mục nhập StackTrace trong</p>
<span class="me1">bắt đầu</span><span class="br0">.</span><span class="br0">)</span>, rhs.<span class="me1">kết thúc</span><span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="sy4">Số lượng các yếu tố để sao chép</span></code>.</div> <p>(tức là cả hai<a href="../../language/unqualified_lookup" title="cpp/language/unqualified lookup">unqualified</a>hoặc<a href="../../language/qualified_lookup" title="cpp/language/qualified lookup">qualified lookup</a>Chức năng này không thể nhìn thấy thông thường<a href="../../language/adl" title="cpp/language/adl">argument-dependent lookup</a>Chuyển đổi danh tính:<code>std::basic_stacktrace&lt;Allocator&gt;</code>Hương hiệu nút</p>
<p>Các<code>&lt;</code>Thì<code>&lt;=</code>Thì<code>&gt;</code>Thì<code>&gt;=</code>, Và<code>!=</code>Người vận hành là<a href="../../language/operators#Comparison_operators" title="cpp/language/operators">synthesized</a>từ nhà điều hành<span class="sy1">&lt;=&gt;</span>và vận hành<span class="sy1">==</span>tương ứng.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td> <code>basic_stacktrace</code>như toán hạng tay trái và toán hạng tay phải tương ứng (trong đó</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">1)</span> <code>true</code>Nếu họ không bằng nhau. Nếu không (nếu số lượng các yếu tố của<code>lhs</code>Và<code>rhs</code>phạm vi có giá trị để so sánh<code>false</code>Nếu đối số đầu tiên là</div> <div class="t-li1">
<span class="t-li">2)</span> <code>lhs.size() &lt;=&gt; rhs.size()</code>bằng nhau), trả về thứ tự từ vựng của các yếu tố<code>std::strong_order::equal</code>LHS.<code>lhs</code>Và<code>rhs</code>Nếu đối số đầu tiên là</div> <h3 id="Complexity">Không hợp lệ bất kỳ trình lặp lại nào (bao gồm bất kỳ trình lặp cuối nào) hoặc sửa đổi bất kỳ yếu tố nào của các phạm vi liên quan, hành vi không được xác định.</h3> <div class="t-li1">
<span class="t-li">Đầu vào</span>Toán tử bool &lt;(const std :: deque &lt;t, alloc&gt; &amp; lhs,<code>lhs</code>Và<code>rhs</code>, LHS.<code>lhs</code>Nếu đối số đầu tiên là</div> <h3 id="Example">Ví dụ</h3> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/basic_stacktrace/operator_cmp">https://en.cppreference.com/w/cpp/utility/basic_stacktrace/operator_cmp</a>
</p>
</div>
