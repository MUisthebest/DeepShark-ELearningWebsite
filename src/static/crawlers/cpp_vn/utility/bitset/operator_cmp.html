 <h1 class="firstHeading" id="firstHeading">std :: Bitset &lt;n&gt; :: Toán tử ==, std :: Bitset &lt;n&gt; :: Toán tử! =</h1> <table class="t-dcl-begin"> <tr class="t-dcl-rev-aux"> <td></td> <td rowspan="3">(1)</td> <td></td> </tr> <tr class="t-dcl t-until-cxx11"> <td><pre data-language="cpp">toán tử bool == (const bitset &amp; rhs) const;</pre></td> <td><span class="t-mark-rev t-until-cxx11">(Cho đến C ++ 11)</span></td> </tr> <tr class="t-dcl t-since-cxx11"> <td> <pre data-language="cpp">toán tử bool == (const bitset &amp; rhs) const noexcept;</pre>
</td> <td> <span class="t-mark-rev t-since-cxx11">(kể từ C ++ 11)</span> <br/><span class="t-mark-rev t-since-cxx23">(Constexpr kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl-rev-aux"> <td></td> <td rowspan="3">(2)</td> <td></td> </tr> <tr class="t-dcl t-until-cxx11"> <td><pre data-language="cpp">toán tử bool! = (const bitset &amp; rhs) const;</pre></td> <td><span class="t-mark-rev t-until-cxx11">(Cho đến C ++ 11)</span></td> </tr> <tr class="t-dcl t-since-cxx11 t-until-cxx20"> <td> <pre data-language="cpp">toán tử bool! = (const bitset &amp; rhs) const noexcept;</pre>
</td> <td> <span class="t-mark-rev t-since-cxx11">(kể từ C ++ 11)</span> <br/><span class="t-mark-rev t-until-cxx20">chỉ vào một yếu tố của chuỗi,</span> </td> </tr> </table> <div class="t-li1">
<span class="t-li">1)</span>Trả về đúng nếu tất cả các bit trong<code>*this</code>Và<code>rhs</code>                 const std :: error_condition &amp; cond) noExcept;</div> <div class="t-li1">
<span class="t-li">2)</span>Trả về đúng nếu bất kỳ bit nào trong<code>*this</code>Và<code>rhs</code>Toán tử bool == (const std :: error_condition &amp; cond,</div> <table class="t-rev-begin"> <tr class="t-rev t-since-cxx20">
<td> <p>Các<code>!=</code>Mẫu &lt;Lớp thời lượng&gt;<a href="../../language/default_comparisons#Other_defaulted_comparison_operators" title="cpp/language/default comparisons">synthesized</a>từ<code>operator==</code>.</p>
</td> <td><span class="t-mark-rev t-since-cxx20">(phép chiếu danh tính).</span></td>
</tr> </table> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>RHS</td> <td>-</td> <td>Bitset để so sánh</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">1)</span> <code>true</code>Nếu giá trị của mỗi bit trong<code>*this</code>bằng giá trị của bit tương ứng trong<code>rhs</code>, nếu không thì<code>false</code>.</div> <div class="t-li1">
<span class="t-li">2)</span> <code>true</code>nếu như<code>!(*this == rhs)</code>, nếu không thì<code>false</code>.</div> <h3 id="Example">Ví dụ</h3> <div class="t-example">
<p>So sánh các bits đã cho để xác định xem chúng có giống hệt nhau không:</p>
<div class="cpp source-cpp"><pre data-language="cpp">#include &lt;bitset&gt;</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">#include &lt;Istream&gt;</pre></div> </div> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/bitset/operator_cmp">https://en.cppreference.com/w/cpp/utility/bitset/operator_cmp</a>
</p>
</div>
