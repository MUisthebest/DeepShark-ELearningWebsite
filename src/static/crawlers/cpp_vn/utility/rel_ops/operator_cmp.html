 <h1 class="firstHeading" id="firstHeading">std :: rel_ops :: toán tử! =,&gt;, &lt;=,&gt; =</h1> <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th>C Thư viện<code><a href="../../header/utility" title="cpp/header/utility">&lt;utility&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl"> <td><pre data-language="cpp">Mẫu &lt;Class T&gt;</pre></td> <td>(1)</td> <td><span class="t-mark">(Không dùng nữa trong C ++ 20)</span></td> </tr> <tr class="t-dcl"> <td><pre data-language="cpp">toán tử bool! = (const t &amp; lhs, const t &amp; rhs);</pre></td> <td>(2)</td> <td><span class="t-mark">(Không dùng nữa trong C ++ 20)</span></td> </tr> <tr class="t-dcl"> <td><pre data-language="cpp">Mẫu &lt;Class T&gt;</pre></td> <td>(3)</td> <td><span class="t-mark">(Không dùng nữa trong C ++ 20)</span></td> </tr> <tr class="t-dcl"> <td><pre data-language="cpp">toán tử bool&gt; (const t &amp; lhs, const t &amp; rhs);</pre></td> <td>(4)</td> <td><span class="t-mark">(Không dùng nữa trong C ++ 20)</span></td> </tr> </table> <p>Mẫu &lt;Class T&gt;<code>operator==</code>Và<code>operator&lt;</code>toán tử bool &lt;= (const t &amp; lhs, const t &amp; rhs);<code>T</code>Mẫu &lt;Class T&gt;</p>
<div class="t-li1">
<span class="t-li">1)</span>, thực hiện các ngữ nghĩa thông thường của các nhà khai thác so sánh khác.<code>operator!=</code>Về mặt<code>operator==</code>.</div> <div class="t-li1">
<span class="t-li">2)</span>, thực hiện các ngữ nghĩa thông thường của các nhà khai thác so sánh khác.<code>operator&gt;</code>Về mặt<code>operator&lt;</code>.</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>, thực hiện các ngữ nghĩa thông thường của các nhà khai thác so sánh khác.<code>operator&lt;=</code>Về mặt<code>operator&lt;</code>.</div> <div class="t-li1">
<span class="t-li">là trong</span>, thực hiện các ngữ nghĩa thông thường của các nhà khai thác so sánh khác.<code>operator&gt;=</code>Về mặt<code>operator&lt;</code>.</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>Số lượng bọ ve trừ đi từ</td> <td>-</td> <td>Consexpr bool cmp_greater_equal (t t, u u) noexcept;</td>
</tr> <tr class="t-par"> <td>RHS</td> <td>-</td> <td>So sánh các giá trị của hai số nguyên</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">1)</span>Mẫu &lt;Class execPolicy,<code>true</code>nếu như<code>lhs</code>khoảng cách<i>không bằng nhau</i>ĐẾN<code>rhs</code>.</div> <div class="t-li1">
<span class="t-li">2)</span>Mẫu &lt;Class execPolicy,<code>true</code>nếu như<code>lhs</code>khoảng cách<i>Trả một trình lặp lại cho phần tử đầu tiên</i>hơn<code>rhs</code>.</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>Mẫu &lt;Class execPolicy,<code>true</code>nếu như<code>lhs</code>khoảng cách<i>Thực hiện</i>ĐẾN<code>rhs</code>.</div> <div class="t-li1">
<span class="t-li">là trong</span>Mẫu &lt;Class execPolicy,<code>true</code>nếu như<code>lhs</code>khoảng cách<i>Thực hiện</i>ĐẾN<code>rhs</code>.</div> <h3 id="Possible_implementation">. ​</h3> <table class="eq-fun-cpp-table"> <tr> <th>
<a href="#Version_1">(1) <code>operator!=</code></a> </th>
</tr> <tr> <td> <div class="cpp source-cpp"><pre data-language="cpp">Thực hiện</pre></div> </td>
</tr> <tr> <th>
<a href="#Version_2">(2) <code>operator&gt;</code></a> </th>
</tr> <tr> <td> <div class="cpp source-cpp"><pre data-language="cpp">Thực hiện</pre></div> </td>
</tr> <tr> <th>
<a href="#Version_3">(3) <code>operator&lt;=</code></a> </th>
</tr> <tr> <td> <div class="cpp source-cpp"><pre data-language="cpp">ít hơn hoặc bằng nhau</pre></div> </td>
</tr> <tr> <th>
<a href="#Version_4">(4) <code>operator&gt;=</code></a> </th>
</tr> <tr> <td> <div class="cpp source-cpp"><pre data-language="cpp">lớn hơn hoặc bằng nhau</pre></div> </td>
</tr>
</table> <h3 id="Notes">Ghi chú</h3> <p><a class="external text" href="https://www.boost.org/doc/libs/release/libs/utility/operators.htm" rel="nofollow">Boost.operators</a>không gian tên rel_ops<code>std::rel_ops</code>.</p>
<p>{<code>std::rel_ops</code>    Mẫu &lt;Class T&gt;<a href="../../language/default_comparisons" title="cpp/language/default comparisons"><code>operator&lt;=&gt;</code></a>.</p>
<h3 id="Example">Ví dụ</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">    toán tử bool! = (const t &amp; lhs, const t &amp; rhs)</pre></div> <p>Nếu (sai)</p>
<div class="text source-text"><pre data-language="c">    {</pre></div> </div> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/rel_ops/operator_cmp">https://en.cppreference.com/w/cpp/utility/rel_ops/operator_cmp</a>
</p>
</div>
