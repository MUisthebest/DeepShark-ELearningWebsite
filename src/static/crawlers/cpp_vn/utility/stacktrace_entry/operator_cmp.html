 <h1 class="firstHeading" id="firstHeading">toán tử ==, toán tử &lt;=&gt;<small>(std :: stacktrace_entry)</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Friend Consexpr Bool Toán tử == (Const StackTrace_Entry &amp; LHS,</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">                                  const stacktrace_entry &amp; rhs) noExcept;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <div class="t-li1">
<span class="t-li">1)</span>void Atomic_flag_wait_explicit (đối tượng Atomic_flag* dễ bay hơi<code>lhs</code>Và<code>rhs</code>                 const std :: pmr :: memory_resource &amp; b) noExcept;<code>stacktrace_entry</code>bạn bè consexpr std :: strong_ordering</div> <div class="t-li1">
<span class="t-li">2)</span>    toán tử &lt;=&gt; (const stacktrace_entry &amp; lhs, const stacktrace_entry &amp; rhs) noExcept;<code>lhs</code>Và<code>rhs</code>Các giá trị bằng nhau khi và chỉ khi chúng đại diện cho cùng một mục nhập StackTrace, hoặc cả hai đều trống.<code>stacktrace_entry</code>Nhận được thứ tự tương đối giữa<code>operator==</code>.</div> <p>Thu được kết quả so sánh ba chiều của<a href="../../language/unqualified_lookup" title="cpp/language/unqualified lookup">unqualified</a>hoặc<a href="../../language/qualified_lookup" title="cpp/language/qualified lookup">qualified lookup</a>Chức năng này không thể nhìn thấy thông thường<a href="../../language/adl" title="cpp/language/adl">argument-dependent lookup</a>Chuyển đổi danh tính:<code>std::stacktrace_entry</code>Hương hiệu nút</p>
<p>Các<code>&lt;</code>Thì<code>&lt;=</code>Thì<code>&gt;</code>Thì<code>&gt;=</code>, Và<code>!=</code>Người vận hành là<a href="../../language/operators#Comparison_operators" title="cpp/language/operators">synthesized</a>từ nhà điều hành<span class="sy1">&lt;=&gt;</span>và vận hành<span class="sy1">==</span>tương ứng.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>E lvalues</td> <td>-</td> <td> <code>stacktrace_entry</code>bằng cách gọi bộ so sánh được lưu trữ</td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">1)</span> <code>true</code>trong tổng số đơn đặt hàng không xác định<code>lhs</code>Và<code>rhs</code>các giá trị phù hợp với mối quan hệ bình đẳng được thiết lập bởi<code>false</code>Nếu đối số đầu tiên là</div> <div class="t-li1">
<span class="t-li">2)</span> <code>std::strong_ordering::equal</code>nếu như<code>lhs</code>Và<code>rhs</code>So sánh bằng nhau.</div> <div class="t-li1">Là xấu nếu<code>std::strong_ordering::less</code>nếu như<code>lhs</code>Trong tổng số đơn đặt hàng nghiêm ngặt.<code>rhs</code>so sánh bằng nhau,</div> <div class="t-li1">Là xấu nếu<code>std::strong_ordering::greater</code>được đặt hàng trước<code>rhs</code>Trong tổng số đơn đặt hàng nghiêm ngặt.<code>lhs</code>(trong trường hợp đó</div> <h3 id="Example">Ví dụ</h3> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator_cmp">https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator_cmp</a>
</p>
</div>
