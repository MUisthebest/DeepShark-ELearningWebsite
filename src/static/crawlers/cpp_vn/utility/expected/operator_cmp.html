 <h1 class="firstHeading" id="firstHeading">toán tử ==<small>std :: hoán đổi</small>
</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;Class T2, Class E2&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> <br/><span class="t-mark">Mẫu &lt;class g&gt;<i>không ngụ ý rằng</i>Consexpr rõ ràng dự kiến ​​(std :: in_place_t, args &amp;&amp; ... args);</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">  Yêu cầu (! STD :: IS_VOID_V &lt;T2&gt;)</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> <br/><span class="t-mark">Mẫu &lt;class ... args&gt;<i>không ngụ ý rằng</i>Consexpr rõ ràng dự kiến ​​(std :: in_place_t, args &amp;&amp; ... args);</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Người bạn của ConstExpr Bool Toán tử == (Const mong đợi &amp; LHS,</pre>
</td> <td>(3)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> <br/><span class="t-mark">Mẫu &lt;class g&gt;<i>không ngụ ý rằng</i>Consexpr rõ ràng dự kiến ​​(std :: in_place_t, args &amp;&amp; ... args);</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">                                  const std :: mong đợi &lt;t2, e2&gt; &amp; rhs);</pre>
</td> <td>(4)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Mẫu &lt;Class T2, Class E2&gt;<code>expected</code>Đối tượng.</p>
<span class="t-li">Đầu vào</span>Constexpr std :: strong_ordering toán tử &lt;=&gt; (const std :: Chrono :: tháng_day &amp; x,<code>expected</code>  Yêu cầu STD :: IS_VOID_V &lt;T2&gt;<code>lhs.has_value()</code>Và<code>rhs.has_value()</code>Người bạn của ConstExpr Bool Toán tử == (Const mong đợi &amp; LHS,<ul>
<li>Mẫu &lt;Class T&gt;<span class="t-v">(1)</span>Friend Consexpr Bool Toán tử == (Const mong đợi &amp; X, Const T2 &amp; Val);<code>*lhs == *rhs</code>Và<code>lhs.error() == rhs.error()</code>Mẫu &lt;Class T2&gt;<code>bool</code>, chương trình là xấu.</li>
<li>Mẫu &lt;Class T&gt;<span class="t-v">(2)</span>Friend Consexpr Bool Toán tử == (Const mong đợi &amp; X, Const T2 &amp; Val);<code>lhs.error() == rhs.error()</code>, nếu các biểu thức<code>bool</code>, chương trình là xấu.</li>
</ul> <span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>void Atomic_flag_wait_explicit (đối tượng Atomic_flag* dễ bay hơi<code>expected</code>người bạn của constexpr bool toán tử == (const mong đợi &amp; x,<code>x</code>                                  const bất ngờ &lt;E2&gt; &amp; E);<code>val</code>.<ul><li>là trung gian nếu cần thiết.<code>*x == val</code>, nếu các biểu thức<code>bool</code>, chương trình là xấu.</li></ul> <span class="t-li">là trong</span>void Atomic_flag_wait_explicit (đối tượng Atomic_flag* dễ bay hơi<code>expected</code>Đối tượng. Các đối tượng so sánh bằng nhau khi và chỉ khi<code>x</code>bằng nhau, và các giá trị chứa là bằng nhau.<code>e.error()</code>.<ul><li>là trung gian nếu cần thiết.<code>x.error() == e.error()</code>, nếu các biểu thức<code>bool</code>, chương trình là xấu.</li></ul> <p>Thu được kết quả so sánh ba chiều của<a href="../../language/unqualified_lookup" title="cpp/language/unqualified lookup">unqualified</a>hoặc<a href="../../language/qualified_lookup" title="cpp/language/qualified lookup">qualified lookup</a>Chức năng này không thể nhìn thấy thông thường<a href="../../language/adl" title="cpp/language/adl">argument-dependent lookup</a>Chuyển đổi danh tính:<code>std::expected&lt;T, E&gt;</code>Hương hiệu nút</p>
<p>Các<code>!=</code>Mẫu &lt;Lớp thời lượng&gt;<a href="../../language/default_comparisons#Other_defaulted_comparison_operators" title="cpp/language/default comparisons">synthesized</a>từ<code>operator==</code>.</p>
<h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>không được hình thành tốt, hoặc nếu kết quả của họ không thể chuyển đổi thành</td> <td>-</td> <td> <code>expected</code>đối tượng để so sánh</td>
</tr> <tr class="t-par"> <td>đánh giá</td> <td>-</td> <td>, nếu các biểu thức<code>x</code> </td>
</tr> <tr class="t-par"> <td>if (auto p = Dynamic_cast &lt;const std :: nested_exception*&gt; (std :: addressof (e))))</td> <td>-</td> <td>không được hình thành tốt, hoặc nếu kết quả của nó không thể chuyển đổi thành<code>x</code> </td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <div class="t-li1">
<span class="t-li">1)</span>Nếu như<code>lhs.has_value() != rhs.has_value()</code>; Nếu không nếu<code>false</code>. Nếu không, nếu<code>lhs.has_value()</code>chứa một giá trị dự kiến ​​và giá trị chứa bằng<code>*lhs == *rhs</code>để định vị<code>lhs.error() == rhs.error()</code>.</div> <div class="t-li1">
<span class="t-li">2)</span>Nếu như<code>lhs.has_value() != rhs.has_value()</code>; Nếu không nếu<code>false</code>. Nếu không, nếu<code>lhs.has_value()</code>chứa một giá trị dự kiến ​​và giá trị chứa bằng<code>true</code>để định vị<code>lhs.error() == rhs.error()</code>.</div> <div class="t-li1">
<span class="t-li">Mẫu &lt;class f, class tuple&gt;</span>Mẫu &lt;Class execPolicy,<code>x.has_value() &amp;&amp; static_cast&lt;bool&gt;(*x == val)</code>.</div> <div class="t-li1">
<span class="t-li">là trong</span>Mẫu &lt;Class execPolicy,<code>!x.has_value() &amp;&amp; static_cast&lt;bool&gt;(x.error() == e.error())</code>.</div> <h3 id="Exceptions">Ngoại lệ</h3> <p>không được hình thành tốt, hoặc nếu kết quả của nó không thể chuyển đổi thành</p>
<h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="unexpected" title="cpp/utility/expected/unexpected"> <span class="t-lines"><span>không ngờ tới</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">Phạm vi :: IOTA</span></span></span></div> </td> <td>được đại diện như một giá trị bất ngờ<br/> <span class="t-mark">Kiểm tra xem một lớp có đại diện cho chính sách thực thi không</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/operator_cmp">https://en.cppreference.com/w/cpp/utility/expected/operator_cmp</a>
</p>
</div>
